File: Codes.python/P12/P12.py
#-------by HYH -------#
import numpy as np
pCan=0.001
pNon=0.999
pPosCan=0.8
pPosNon=0.1
z='positive'
if 'positive'==z:
	p=[pPosCan*pCan,pPosNon*pNon]
else:
	p=[(1-pPosCan)*pCan,(1-pPosNon)*pNon]
p=p/np.sum(p)
print('The probability of having cancer given the %s test:\n'% z,'\n',p[0])
print('The probability of cancer free given the %s test:\n'%z,'\n',p[1])

File: Code-C++/P12/P12/P12.cpp
// P12.cpp : 定义控制台应用程序的入口点。
//

#include "stdafx.h"
#include <iostream>
 #include <string>　
using namespace std;


int main()
{
	float pCan=0.001;
	float pNon=0.999;
	float pPosCan=0.8;
	float pPosNon=0.1;
	float p[2];
	string str;
	cin>>str;
	if(str=="positive")
	{
		float p1=pPosCan*pCan;
		float p2=pPosNon * pNon;
		float P1=p1/(p1+p2);
		float P2=p2/(p1+p2);
		cout<<"The probability of having cancer given the "<<P1<<"   ";
		cout<<P2<<endl;
	}
	else
	{
		float p1=(1-pPosCan)*pCan;
		float p2=(1-pPosNon)*pNon;
		float P1=p1/(p1+p2);
		float P2=p2/(p1+p2);
		cout<<"The probability of having cancer given the "<<P1<<"   ";
		cout<<P2<<endl;
	}
	return 0;
}

